python/python/bystro/tada_mixture/mixture_tada_em.py

```diff
@@ -374,6 +427,101 @@ def predict_logproba(self, data):
                 log_proba[i, k] = np.sum(probs)
         return log_proba
 
+    def _store_instance_variables(self, trainable_variables):
+        """
+        Saves the learned variables
+
+        Parameters
+        ----------
+        trainable_variables : list
+            List of variables to save
+        """
+        self.pi = trainable_variables[0]
+        Lambda_list_l = trainable_variables[1]
+        alpha_ls = trainable_variables[2]
+        sigmoid = nn.Sigmoid()
+        self.Lambda = np.zeros((self.K,self.p))
+        self.Alpha = np.zeros((self.K,self.p))
+        for k in range(self.K):
+            Lambda_k = Lambda_list_l[k]
+            alpha_k = sigmoid(alpha_ls[k])
+
+            self.Lambda[k] = Lambda_k.detach().numpy()
+            self.Alpha[k] = alpha_k.detach().numpy()
+
+    def _initialize_variables(self, X):
+        """
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        """
+        model = GaussianMixture(self.K)
+        model.fit(X)
+        Lambda_list_l = []
+        for i in range(self.K):
+            Lambda_list_l.append(
+                torch.tensor(
+                    1.2 * model.means_[i].astype(np.float32) + 0.1,
```

Comment:
here and throughout, I would recommend pulling out and naming magic numbers like these